### DISCUSSION TITLE
How to enable installation of Eigen

### DISCUSSION THREAD

**Discussion by andrewlhicks** [firedrakeproject/firedrake#2940] [Firedrake support] [answered]
I just updated Firedrake to the latest version and I immediately started having issues with an op2.Kernel object. In my kernel string I open with this line:
```
#include <Eigen/Dense>
```
And this leads to the following error when compiling:
```
/scratch/ahick17/firedrake/.cache/pyop2/c9/57ce217894e50cbc2c31d4cbe1d1df_p3949555.cpp:4:10: fatal error: Eigen/Dense: No such file or directory
    4 | #include <Eigen/Dense>
      |          ^~~~~~~~~~~~~
compilation terminated.
```
After looking at the firedrake-update text output, it was quite clear that this time, unlike in former times when updating Firedrake, no mention of Eigen was ever made. So it seems to me that this was intentionally removed from the default options for Firedrake. I do need it for my kernel, however.

My question is, what's the simplest way to fix my Firedrake installation to include Eigen?

**Comment by connorjward** [comment, marked as answer]
Eigen was recently removed from the default PETSc build that Firedrake uses. Running `export PETSC_CONFIGURE_OPTIONS=--download-eigen` prior to `firedrake-install` should get it back for you.

**Reply by andrewlhicks** [reply]
This seems to work fine, thanks!

One question though, was there a reason why it was decided that Eigen should be removed from the default PETSc build?

**Reply by connorjward** [reply]
Eigen was only used in Firedrake for the COFFEE backend to Slate. We recently dropped support for COFFEE in favour of loopy and this allowed us to get rid of the old Slate backend, including Eigen.
